server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/RequestScopedIdpDefinitionsForZone.java

```diff
@@ -0,0 +1,22 @@
+package org.cloudfoundry.identity.uaa.provider.saml;
+
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.cloudfoundry.identity.uaa.provider.IdentityProvider;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.annotation.RequestScope;
+
+@RequestScope
+@Component("requestScopedIdpDefinitionsForZone")
+public class RequestScopedIdpDefinitionsForZone {
+    private final ConcurrentHashMap<String, List<IdentityProvider>> zoneIdps = new ConcurrentHashMap<>();
```

Comment:
I agree that using ConcurrentHashMap instead of Caffeine is fine for this simple caching scenario.  Caffeine is a loading cache and is better used when cache needs to be refreshed and synchronized frequently.  
ConcurrentHashMap is also used by spring as a default cache implementation if no other library is provided: https://docs.spring.io/spring-framework/docs/5.3.18/reference/html/integration.html#cache-store-configuration-jdk